staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -2198,3 +2212,18 @@ func hasAdditionalNewField(original, modified map[string]interface{}) bool {
 	}
 	return false
 }
+
+func setMergeKeyForEmptyMapInSlice(m []interface{}, mergeKey string) error {
+	for _, v := range m {
+		typedV, ok := v.(map[string]interface{})
+		if !ok {
+			return mergepatch.ErrBadArgType(typedV, v)
+		}
+
+		if len(typedV) == 0 {
+			typedV[mergeKey] = nil
```

Comment:
is setting the value to an explicit nil value valid? doesn't that fail validation for non-nullable fields in a custom resource?

Reply:
To my understand, strategic patch is unsupported for custom resource, In native k8s resource, all `patchMergeKey` types are string, so it's safe to set to nil